File: Week-2/1-caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    // Validate key
    int i = 0;
    string s = argv[1];
    char c = s[i];  
    while (c != '\0')
    {
        if (!isdigit(c))
        {
            printf("Usage: ./caesar key\n");
            return 1;
        }
        c = s[++i];
    }
    // Convert key to int
    int key = atoi(s) % 26;
    // Check full shift case
    if (key > 26)
    {
        key = key % 26;
    }
    // Ask user for a plain text
    string plain = get_string("plaintext: ");
    // Encrypt plain text using user key
    c = plain[0];
    i = 0;
    printf("ciphertext: ");
    while (c != '\0')
    {
        char ck = c;
        if (c >= 'a' && c <= 'z')
        {
            // Check lower case
            ck = (c + key) > 'z' ? 'a' + (c + key - 1) - 'z' : (c + key);
        } 
        else if (c >= 'A' && c <= 'Z')
        {
            // Check upper case
            ck = (c + key) > 'Z' ? 'A' + (c + key - 1) - 'Z' : (c + key);
        } 
        printf("%c", ck);
        // Pick next char
        c = plain[++i];
    }
    printf("\n");
}

File: Week-6/5 - caesar/caesar.py
from sys import argv, exit
from cs50 import get_string


def encrypt(c, k):
    code = ord(c)
    shift = code + k
    if code >= 65 and code <= 90:
        # Uppercase
        return str(chr(shift)) if shift <= 90 else str(chr(65 + (shift - 65) % 26))
    elif code >= 97 and code <= 122:
        # Lowercase
        return str(chr(shift)) if shift <= 122 else str(chr(97 + (shift - 97) % 26))
    else:
        return str(chr(code))


# Validate argv
if len(argv) != 2:
    print("Usage: python caesar.py k")
    exit(1)
# Ask user for text
text = get_string("plaintext:  ")
# Read command line arg
k = int(argv[1])
# Encrypt text
encrypted = ""
for c in text:
    encrypted += encrypt(c, k)
# Print encrypted text
print(f"ciphertext: {encrypted}")